assignments/HW4.md

```diff
@@ -0,0 +1,194 @@
+
+You are correctly skeptical about the effort required to pivot the product in this new direction, but fortunately the somewhat disconnected microservice demand gives you a way to explore the idea without requiring a major up front change in the existing codebase. Instead, you’ll (1) develop this potential feature in isolation by training a machine learning model that predicts potential applicant success and deploying it as a microservice, and then (2) reason about, document, and choose between alternative designs that would allow you to integrate it into the existing codebase. Finally, you will (3) individually consider the ethical implications of the feature, and what questions you truly need to ask in QA to satisfy possible ethical concerns.
+
+## Starter code, data, and repository
+
+Create a new team repository by following this link for your team, which will clone starter data and code:
+
+Placeholder: https://classroom.github.com/a/bRnF4uC8
+
+There is an example flask-based app in the `apps` directory. Use this as a template to build and deploy the microservice. You may wish to use this article and this repository as resources on building and saving a scikit-learn model, and then deploying it using flask.
+**MAKE SURE** to read the `README.MD` of the repository to learn about and use Pipenv, as it will allow for hassle-free collaboration.
+
+This starter code uses arguments from the query string (specifically age, health, and absences) to query the model. This can/should be expanded or replaced as a part of your implementation (note that the API also accepts JSON requests).
+
+Note that the default flask service in the starter code should be run, and is accessible, on port 80. Feel free to change as needed.
+
+## Deadlines and deliverables
+
+This homework has four (4) deadlines and four (4) deliverables. The first deadline (Thursday, Oct 27th) is a checkpoint to make sure you are succeeding at building your microservice-based ML model. The second deadline (Tuesday, Nov 1st) is for the complete trained model deployable as a microservice. The third deadline (Thursday, Nov 3rd) is for a design document explaining how you would integrate the new feature into your existing codebase. The fourth deadline (Tuesday, Nov 8th) is for one (1) individual document making an ethical argument about the feature.
+
+### (1) Development checkpoint - (due Thursday, Oct 27th, 23:59) | 5 points
```

Comment:
Given that this is also an API design exercise, there's no enough time for this activity, IMO

Reply:
Are you referring to the 1st development checkpoint?

Reply:
I thought just defining endpoints of the API could be done in 4 days (Monday~Thurs).

Reply:
pushed this back to the tuesday after